### PR TITLE
Fix and test route snapping, places autocomplete, tracking

### PR DESCRIPTION
**Pull request by zeladada**
And added tests

### CODE REVIEW

File: scripts/run_ghopper.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# grep -E -i ".*(\[main\]|\[INFO\]|erro?r?|exception|at.*(java|graphhopper)|exit*).*" # grep whitelist lines
+# grep -E -i "^((?!download|points?|maven).)*$" # grep blacklist lines
```
Comment (young-k) [inline scripts/run_ghopper.sh]: delete the comments

File: src/test/routes.test.js [hunk 2]
```diff
@@ -18,78 +58,181 @@ beforeAll(async () => init.then((res) => {
...
+                await request(server).get(delayReq).expect((delayRes) => {
+                    console.log(delayRes.body);
+                    checkDelayResponseValid(delayRes);
+                });
+            }));
+        });
+    });
+    */
 
     // TODO
```
Comment (young-k) [inline src/test/routes.test.js]: delete the TODO?

File: src/test/routes.test.js [hunk 3]
```diff
@@ -18,78 +58,181 @@ beforeAll(async () => init.then((res) => {
...
+        checkDelayResponseValid(res);
     }));
 
-    const afternoonToday = moment().endOf('day').subtract(12, 'hours').unix();
-    const route3 = `?start=42.444759,-76.484183&end=42.442503,-76.485845&time=${afternoonToday}&arriveBy=false&destinationName="Schwartz"`;
-    test(`afternoon today: ${route}${route3}`, () => request(server).get(route + route3).expect((res) => {
-        checkRouteValid(res);
-    }));
+    // not needed, use route data from before
+    /*
```
Comment (young-k) [inline src/test/routes.test.js]: delete

File: src/test/routes.test.js (hunk 2 above)
Comment (zeladada) [reply src/test/routes.test.js]: It's just to signal we need to write more tests, I'm going to leave it.

**Pull request closed without merging**